index.html

```diff
@@ -2531,22 +2588,48 @@ <h3>Algorithm</h3>
         <li>If <var>active property</var> has a <a>type mapping</a> in
           <var>active context</var>,
           <span class="changed">other than <code>@id</code>, <code>@vocab</code>, or <code>@none</code>,</span>
-          add an <code>@type</code> <a>entry</a> to
+          add <code>@type</code> to
           <var>result</var> and set its value to the value associated with the
           <a>type mapping</a>.</li>
         <li>Otherwise, if <var>value</var> is a <a>string</a>:
           <ol>
-            <li>If a <a>language mapping</a> is associated with
+            <li class="changed">If both a non-`null` <a>language mapping</a>
+              and a non-`null` <a>direction mapping</a> are associated with
+              <var>active property</var> in <var>active context</var>,
+              add <code>@language</code> to <var>result</var> and set its
+              value to the <a>language tag</a> associated with the
+              <a>language mapping</a>.
+              Add a <code>@direction</code> to <var>result</var> and set its
+              value to the <a>direction mapping</a>.</li>
+            </li>
+            <li class="changed">Otherwise, if a non-`null` <a>language mapping</a> is associated with
               <var>active property</var> in <var>active context</var>,
-              add an <code>@language</code> to <var>result</var> and set its
-              value to the language code associated with the
-              <a>language mapping</a>; unless the
-              <a>language mapping</a> is set to <code>null</code> in
+              add <code>@language</code> to <var>result</var> and set its
+              value to the <a>language tag</a> associated with the <a>language mapping</a>.
+              Add a <code>@direction</code> to <var>result</var> and set its
+              value to the <a>default base direction</a> from <a>active context</a>; unless the
+              <a>direction mapping</a> is set to <code>null</code> in
               which case no such <a>entry</a> is added.</li>
-            <li>Otherwise, if the <var>active context</var> has a
-              <a>default language</a>, add an <code>@language</code>
+            </li>
+            <li class="changed">Otherwise, if a non-`null` <a>direction mapping</a> is associated with
+              <var>active property</var> in <var>active context</var>,
+              add <code>@language</code> to <var>result</var> and set its
+              value to the <a>default language</a> from <a>active context</a>,
+              if <a>active context</a> has a <a>default language</a>,
+              and the <a>language mapping</a> is not `null`.
+              Add a <code>@direction</code> to <var>result</var> and set its
+              value to the <a>direction mapping</a>; unless the
+              <a>direction mapping</a> is set to <code>null</code> in
+              which case no such <a>entry</a> is added.</li>
```

Comment:
"unless the direction mapping is set to null..." is useless here, as the condition at the start of the step prevents it from being null.

Reply:
Simplified Value Expansion along the lines of Value Compaction.